### ISSUE TITLE
[FEAT] Add --exclude-topics flag to filter repos by topics

### ISSUE THREAD

**Issue by fuleinist** [prakhar1605/Opencollab-mcp#1]
Problem: Users may want to avoid repos related to certain topics (e.g., avoid "tensorflow" or "android" if they dont work in those domains).

Solution: Add an optional `--exclude-topics` flag to filter out repos that have certain topics.

Use case: A user who knows Python but wants to avoid ML/AI repos can filter them out.

Example: `opencollab_find_issues --exclude-topics machine-learning,torch` would filter out repos tagged with those topics.

This is a small, well-scoped enhancement that adds one new flag parameter.

**Comment by Sujith-RMD** [comment]
Hey @prakhar1605! I'd like to pick this up.
Was thinking of adding an exclude_topics param to the relevant input model and filtering repos in discovery.py before they're returned. Will also cover it with a test.
Can you assign this to me?

**Comment by prakhar1605** [comment]
Thanks for the proposal @fuleinist — this is a genuinely useful feature. A Python dev who doesn't do ML asking for "good first issues" gets buried in TensorFlow/PyTorch results today, and there's no clean way around it. 👍

And @Sujith-RMD — assigning this to you. The plan you outlined is exactly right. A few implementation notes that'll save you time:

**1. It's a Pydantic field, not a CLI flag.**

Quick context on the architecture: OpenCollab is an MCP server, so tools are called by the AI client with JSON arguments rather than CLI flags. The user-facing experience is *"find me Python issues, but skip anything tagged machine-learning or pytorch"* — the AI then constructs:

```json
{"language": "Python", "exclude_topics": ["machine-learning", "pytorch"]}
```

So the change lives in `src/opencollab_mcp/models.py` — add an optional field to `LanguageInput`:

```python
exclude_topics: list[str] | None = Field(
    default=None,
    description="Topics to exclude from results (e.g. ['machine-learning', 'pytorch'])",
)
```

**2. Filter at the right layer.**

The four discovery tools that should respect this: `find_issues`, `trending_repos`, `find_mentor_repos`, `weekend_issues`. Two of them (`trending_repos`, `find_mentor_repos`) already deal in repo objects with a `topics` field, so filtering is a one-line list comprehension. For `find_issues` and `weekend_issues` (which return *issues*, not repos), the issue payload doesn't include topics — you'd need to skip them in this PR and just note it as a follow-up, or do an extra repo lookup per issue (which I'd push back on for rate-limit reasons). Pick the simpler path.

**3. Normalize the comparison.**

GitHub topics are lowercase and dash-separated (`machine-learning`, not `Machine Learning`). Add a `[t.lower().strip() for t in (exclude_topics or [])]` so users don't get tripped up by case.

**4. Tests.**

Please add at least two cases to `tests/test_tools.py`:
- A trending_repos call where `exclude_topics=["machine-learning"]` filters out an ML repo from the mocked response
- A trending_repos call with `exclude_topics=None` (the default) returns everything unchanged

The `mock_github` fixture in `tests/conftest.py` is what you want.

**5. Branch protection note.**

`main` is protected — please open a feature branch (`feat/exclude-topics`) and PR into `main`. CI runs ruff + pytest on Python 3.10/3.11/3.12 and all three need to pass.

Take your time, no rush. Tag me on the PR when it's ready or if anything is unclear.

**Comment by albin-george-kurian** [comment]
Hi @prakhar1605, 
Is this still open? If so I would like to work on it. I have a few approaches in mind for implementing it

**Comment by prakhar1605** [comment]
yes do it @albin-george-kurian
